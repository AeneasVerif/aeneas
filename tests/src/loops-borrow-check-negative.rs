//@ [!borrow-check] skip
//@ [borrow-check] known-failure

// This succeeds
fn loop_a<'a>(a: &'a mut u32, b: &'a mut u32) -> &'a mut u32 {
    let mut x = 0;
    while x > 0 {
        x += 1;
    }
    if x > 0 {
        a
    } else {
        b
    }
}

// This fails
fn loop_a_b<'a, 'b>(a: &'a mut u32, b: &'b mut u32) -> &'a mut u32 {
    let mut x = 0;
    while x > 0 {
        x += 1;
    }
    if x > 0 {
        a
    } else {
        b // Expect lifetime 'a
    }
}
