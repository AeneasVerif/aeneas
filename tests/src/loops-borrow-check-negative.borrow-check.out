[[92mInfo[39m ] Imported: tests/llbc/loops_borrow_check_negative.llbc
[[91mError[39m] Can't end abstraction 16 as it is set as non-endable
Source: 'tests/src/loops-borrow-check-negative.rs', lines 18:0-18:66
